elasticsearch/connection/http_urllib3.py

```diff
@@ -152,6 +159,8 @@ def perform_request(self, method, url, params=None, body=None, timeout=None, ign
 
             request_headers = self.headers
             if headers:
+                if self.http_compress == True:
```

Comment:
this needs to be outside of the `if headers` block, otherwise it will only be used if custome headers are specified.

if should be:
```python
if self.http_compress and body:
    body = gzip.compress(body)
```

Reply:
Derp! Updating.

Reply:
good eye @HonzaKral